### PR TITLE
Sync packages for WP 6.1 RC 2

### PR DESCRIPTION
**Pull request by ockham**
Includes the latest round of Gutenberg bugfix PRs that were approved for inclusion during [yesterday’s triage session](https://wordpress.slack.com/archives/C02QB2JS7/p1666019901292909). The following Gutenberg PRs were cherry-picked (see the corresponding [GB PR](https://github.com/WordPress/gutenberg/pull/45051)):

- https://github.com/WordPress/gutenberg/pull/45041
- https://github.com/WordPress/gutenberg/pull/45045
- https://github.com/WordPress/gutenberg/pull/44999
- https://github.com/WordPress/gutenberg/pull/44972
- https://github.com/WordPress/gutenberg/pull/44858
- https://github.com/WordPress/gutenberg/pull/44878
- https://github.com/WordPress/gutenberg/pull/44809
- https://github.com/WordPress/gutenberg/pull/44864
- https://github.com/WordPress/gutenberg/pull/44853
- https://github.com/WordPress/gutenberg/pull/45050
- https://github.com/WordPress/gutenberg/pull/44887

Needed for today's WP 6.1 RC 2 release.

Trac ticket: https://core.trac.wordpress.org/ticket/56467

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/blocks/search.php [hunk 1]
```diff
@@ -367,12 +367,12 @@ function styles_for_block_core_search( $attributes ) {
...
 	$has_text_color = ! empty( $attributes['style']['color']['text'] );
 	if ( $has_text_color ) {
-		$button_styles[] = sprintf( 'color: %s;', esc_attr( $attributes['style']['color']['text'] ) );
+		$button_styles[] = sprintf( 'color: %s;', $attributes['style']['color']['text'] );
 	}
 
 	$has_background_color = ! empty( $attributes['style']['color']['background'] );
 	if ( $has_background_color ) {
-		$button_styles[] = sprintf( 'background-color: %s;', esc_attr( $attributes['style']['color']['background'] ) );
+		$button_styles[] = sprintf( 'background-color: %s;', $attributes['style']['color']['background'] );
```
Comment (audrasjb) [inline src/wp-includes/blocks/search.php:375]: Is it intended? Should we replace `esc_attr` with `esc_html` instead?

**Review by audrasjb — COMMENTED** [review]
_[Committer double signoff]_

I just have one question, otherwise the PR looks good to me.

File: src/wp-includes/blocks/search.php (hunk 1 above)
Comment (ockham) [reply src/wp-includes/blocks/search.php:375]: Yeah, it's intended. We're not really removing the `esc_attr` but moving to the [latest possible point](https://github.com/WordPress/wordpress-develop/pull/3485/files#diff-b2e87a1e901a51c9ba26e81131bac3c18258a1c4310c21a276f2d98598d96c95R403). (So I don't think we should add `esc_html` here.)

FWIW, this change was part of yesterday's 6.0.3 (and earlier versions) security release.

Comment (audrasjb) [reply src/wp-includes/blocks/search.php:375]: Alright, we're good to go then!

**Review by audrasjb — APPROVED** [review]
_[double committer signoff]_
Good to go!

**Review by ockham — COMMENTED** [review]
_[Committer double signoff]_

(Also signing off on this in case it's needed. I hope this is how it works, since I'm also the PR author. FWIW, the backported PRs are all by other folks.)

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r54632](https://core.trac.wordpress.org/changeset/54632).

**Pull request closed without merging**